Sources/SyntaxAnalysis/DeclarationSyntaxVisitor.swift

```diff
@@ -517,3 +517,64 @@ public final class DeclarationSyntaxVisitor: PeripherySyntaxVisitor {
             }
     }
 }
+
+// MARK: - Inline Comment Commands
+
+private extension SyntaxProtocol {
+    /// All trivia where inline comment commands can be found.
+    ///
+    /// Matches uses like these:
+    ///
+    /// ```
+    /// // periphery:ignore
+    /// Foo {
+    /// }
+    /// ```
+    ///
+    /// ```
+    /// Foo { // periphery:ignore
+    /// }
+    /// ```
+    ///
+    /// ```
+    /// Foo {} // periphery:ignore
+    /// ```
+    var commentCommandTrivia: Trivia {
+        var commandTrivia = leadingTrivia
+
+        if let hasMemberBlock = self as? HasMemberBlock {
+            commandTrivia = commandTrivia
+                .merging(hasMemberBlock.memberBlock.leftBrace.trailingTrivia)
+                .merging(hasMemberBlock.memberBlock.rightBrace.trailingTrivia)
+        }
+
+        if let hasCodeBody = self as? HasCodeBody, let body = hasCodeBody.body {
+            commandTrivia = commandTrivia
+                .merging(body.leftBrace.trailingTrivia)
+                .merging(body.rightBrace.trailingTrivia)
+        }
+
+        return commandTrivia
+    }
+}
+
+/// Identifies types with a MemberBlockSyntax child
+private protocol HasMemberBlock {
+    var memberBlock: MemberBlockSyntax { get }
+}
+
+extension ActorDeclSyntax: HasMemberBlock {}
+extension ClassDeclSyntax: HasMemberBlock {}
+extension ProtocolDeclSyntax: HasMemberBlock {}
+extension ExtensionDeclSyntax: HasMemberBlock {}
+extension StructDeclSyntax: HasMemberBlock {}
+extension EnumDeclSyntax: HasMemberBlock {}
+
+/// Identifies types with a CodeBlockSyntax child
+private protocol HasCodeBody {
```

Comment:
Similar here but for [`HasTrailingCodeBlock`](https://github.com/swiftlang/swift-syntax/blob/main/Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift#L51)